### DISCUSSION TITLE
Can http QUERY method be used in helidon (nima)

### DISCUSSION THREAD

**Discussion by hrstoyanov** [helidon-io/helidon#7320] [General]
https://httpwg.org/http-extensions/draft-ietf-httpbis-safe-method-w-body.html

**Comment by romain-grecourt** [comment]
The API allows the use of any method name:

E.g.

```java
WebServer.builder()
        .routing(rules -> rules
                .route(Http.Method.create("QUERY"), (req, res) -> {
                    String query = req.content().as(String.class);
                    String result = "..."; 
                    res.send(result);
                }))
        .build()
        .start();
```

We can add explicit support for this in the Helidon APIs without breaking backward compatibility.

However, this depends on the proposal getting finalized as an RFC, and also depends on wether the Helidon team agrees to include such support.

**Reply by hrstoyanov** [reply]
Thanks, I will use the code you provided, this is  99.999% guaranteed to make it as RFC, keep an eye on it.
